src/algorithms/SortingAlgorithms.java

```diff
@@ -3,45 +3,136 @@
+    public static void sortBySelectionSorting(int[] array) {
+        for (int k = 0; k < array.length; k++) {
+            for (int l = k + 1; l < array.length; l++) {
+                if (array[l] < array[k]) {
+                    swap(array, k, l);
+                }
+            }
+        }
+    }
 
+    public static void sortByInsertionSorting(int[] array) {
+        for (int k =0; k < array.length; ) {
+            for (int l = k + 1; l < array.length; l++) {
+                if (array[l] < array[k]) {
+                    swap(array,k,l);
+                       int m=k;
+                  while(m>0 && array[m-1]>array[m]){
+                    swap(array,m-1,m);
+                      m--;
+                  }
+                }
+                 k++;
+            }
+        }
     }
 
-    public static void sortByInsertionSorting(int[] array){
+        public static void sortByMergeSorting ( int[] array) {
+            if(array.length<2){
+                return;
+            }
+               int middle = array.length / 2;
+               int[] leftArr = new int[middle];
+               int[] rightArr = new int[array.length - middle];
+                  for (int i = 0; i < middle; i++) {
+                      leftArr[i] = array[i];
+                  }
+                      for (int j = 0; j < rightArr.length; j++) {
+                          rightArr[j] = array[middle + j];
+                      }
+            sortByMergeSorting(leftArr);
+           sortByMergeSorting(rightArr);
+          mergeArray(array, leftArr, rightArr);
+        }
+            public static void mergeArray(int [] array, int [] leftArr, int []rightArr ){
+                int i = 0;
+                int j = 0;
+                int k = 0;
+                  while (i < leftArr.length && j < rightArr.length) {
+                       if (leftArr[i] <= rightArr[j]) {
+                        array[k] = leftArr[i];
+                        i++;
+                       } else { array[k] = rightArr[j];
+                                j++;
+                         }
+                    k++;
+                  }
 
-    }
+                while (i < leftArr.length) {
+                    array[k] = leftArr[i];
+                     i++;
+                      k++;
+                }
 
-    public static void sortByMergeSorting(int[] array){
+                while (j < rightArr.length) {
+                    array[k] = rightArr[j];
+                     j++;
+                      k++;
+                }
 
-    }
+            }
 
-    public static void sortByQuickSorting(int[] array){
+            public static void sortByQuickSorting ( int[] array, int begin, int end) {
 
-    }
+                if(begin>=end){return;}
+                int p = partition(array, begin, end);
+       if(begin<p-1){
+            sortByQuickSorting(array, begin, p - 1);}
+       if(p<end){
+            sortByQuickSorting(array, p + 1, end);}
```

Comment:
In order for the code to be readable, you can format the code using **Ctrl+Alt+L**

Reply:
okay thanks